pjsip/include/pjsip-simple/dlg_event.h

```diff
@@ -0,0 +1,274 @@
+ * framework (evsub.h) and extends the framework by implementing
+ * "dialog-info+xml" event package. This could be useful for features such as
+ * Busy Lamp Field (BLF).
+ */
+
+
+/**
+ * Initialize the dialog event module and register it as endpoint module and
+ * package to the event subscription module.
+ *
+ * @param endpt     The endpoint instance.
+ * @param mod_evsub The event subscription module instance.
+ *
+ * @return          PJ_SUCCESS if the module is successfully
+ *                  initialized and registered to both endpoint
+ *                  and the event subscription module.
+ */
+PJ_DECL(pj_status_t) pjsip_dlg_event_init_module(pjsip_endpoint *endpt,
+                                                 pjsip_module *mod_evsub);
+
+
+/**
+ * Get the dialog event module instance.
+ *
+ * @return          The dialog event module instance.
+ */
+PJ_DECL(pjsip_module*) pjsip_bdlg_event_instance(void);
+
+
+/**
+ * Maximum dialog event status info items which can handled by application.
+ *
+ */
+#define PJSIP_DLG_EVENT_STATUS_MAX_INFO  8
+
+
+/**
+ * This structure describes dialog event status of an entity.
+ */
+struct pjsip_dlg_event_status
+{
+    unsigned        info_cnt;             /**< Number of info in the status */
+    struct {
+
+        pj_str_t    dialog_info_state;    /**< Dialog-Info state            */
+        pj_str_t    dialog_info_entity;   /**< Dialog-Info entity           */
+        pj_str_t    dialog_call_id;       /**< Dialog's call_id             */
+        pj_str_t    dialog_remote_tag;    /**< Dialog's remote-tag          */
+        pj_str_t    dialog_local_tag;     /**< Dialog's local-tag           */
+        pj_str_t    dialog_direction;     /**< Dialog's direction           */
+        pj_str_t    dialog_id;            /**< Dialog's id                  */
+        pj_str_t    dialog_state;         /**< Dialog state                 */
+        pj_str_t    dialog_duration;      /**< Dialog duration              */
+
+        pj_xml_node *dialog_node;         /**< Pointer to tuple XML node of
+                                               parsed dialog-info body received
+                                               from remote agent. Only valid
+                                               for client subscription. If the
+                                               last received NOTIFY request
+                                               does not contain any dialog-info
+                                               body, this will be set to NULL*/
+        pj_str_t    local_identity;
+        pj_str_t    local_identity_display;
+        pj_str_t    local_target_uri;
+
+        pj_str_t    remote_identity;
+        pj_str_t    remote_identity_display;
+        pj_str_t    remote_target_uri;
+
+    } info[PJSIP_DLG_EVENT_STATUS_MAX_INFO]; /**< Array of info.            */
+
+    pj_bool_t       _is_valid;            /**< Internal flag.               */
```

Comment:
This flag seems to be used by `tmp_status`, perhaps better put it inside the `pjsip_dlg_event` instead?

Reply:
This is also what we do in presence.